wagtail/snippets/tests/test_viewset.py

```diff
@@ -164,6 +164,19 @@ def test_deconstruct(self):
         # It should not add any extra kwargs for the icon
         self.assertEqual(kwargs, {"required": False})
 
+    def test_snippet_icon_override_if_custom_icon_is_defined(self):
+        # Snippet icon is overridden if it is defined, FullFeaturedSnippet has a custom icon as "cog"
+        chooser_block = SnippetChooserBlock(FullFeaturedSnippet)
+        self.assertEqual(chooser_block.target_model.snippet_viewset.icon, "cog")
+
+    def test_snippet_icon_override_if_custom_icon_is_not_defined(self):
+        mock_viewset = mock.MagicMock()
+        mock_model = mock.MagicMock()
+        mock_model.snippet_viewset = mock_viewset
```

Comment:
Suggestion: It would be much nicer if these could use existing snippets in the project instead. There are plenty of snippets included which don't set an explicit icon.